supervisor/homeassistant/core.py

```diff
@@ -450,15 +454,25 @@ async def _block_till_run(self, version: AwesomeVersion) -> None:
                 break
 
             # 2: Check if API response
-            if await self.sys_homeassistant.api.check_api_state():
+            if await self.sys_homeassistant.api.check_api_state(
+                check_running=check_running
+            ):
+                if not check_running:
+                    # API initially available, move deadline up and check API
+                    # state to be running now
+                    _LOGGER.info("Detect a Home Assistant instance")
+                    deadline = datetime.now() + STARTUP_API_CHECK_RUNNING_TIMEOUT
+                    check_running = True
+                    continue
+
                 _LOGGER.info("Detect a running Home Assistant instance")
                 self._error_state = False
                 return
 
         self._error_state = True
         if timeout:
             raise HomeAssistantStartupTimeout(
-                "No API response in 5 minutes, assuming core has had a fatal startup error",
+                "No Home Assistant Core response, assuming a fatal startup error",
```

Comment:
Maybe change `check_api_state` to `get_api_state` (and make the existing function a wrapper around it) so we can log the state that core was in here which would give us a better hint as to what went wrong if we have a problem here in the future